src/test/python/testcollections.py

```diff
@@ -0,0 +1,52 @@
+import find_projection
+
+import _pyrt
+import unittest
+import asyncio
+
+class TestCollections(unittest.TestCase):
+
+    def test_stringmap(self):
+        m = _pyrt.StringMap()
+        m.Insert("hello", "world")
+
+        self.assertTrue(m.HasKey("hello"))
+        self.assertFalse(m.HasKey("world"))
+        self.assertEqual(m.get_Size(), 1)
+        self.assertEqual(m.Lookup("hello"), "world")
+
+
+    def test_stringmap_changed_event(self):
+        loop = asyncio.new_event_loop()
+        asyncio.set_event_loop(None)
+        
+        async def async_test():
+            future = loop.create_future()
+
+            def onMapChanged(sender, args): 
+                self.assertEqual(args.get_CollectionChange(), 1)
+                self.assertEqual(args.get_Key(), "dr")
+
+                self.assertEqual(sender.get_Size(), 2)
+                self.assertTrue(sender.HasKey("dr"))
+                self.assertTrue(sender.HasKey("hello"))
+                
+                loop.call_soon_threadsafe(asyncio.Future.set_result, future, True)
+
+            m = _pyrt.StringMap()
+            m.Insert("hello", "world")
+            token = m.add_MapChanged(onMapChanged)
+            m.Insert("dr", "who")

```

Comment:
The callback should trigger here synchronously as well, right? Or does it only work in an async context?

Reply:
Events aren't guaranteed to come in on the same thread, though i think this one might. However, I wrote this test asynchronously so I could await on the event firing. Let me know if there's a more pythonic way of doing that